tcms/telemetry/api.py

```diff
@@ -156,6 +158,7 @@ def execution_trends(query=None):
 
         else:
             _append_status_counts_to_result(count, data_set)
+            data_set['TOTAL'].append(TestExecution.objects.filter(run_id=run_id).count())
```

Comment:
The `count` variable already holds all of the statuses so you can calculate the total based on it (instead of quering it from the DB) and perform the calculation inside `_append_status_count_to_result()`.

Reply:
Another reason to perform the calculations in `_append_status_count_to_result()` is that this function is called twice, see line 168 below. As-is the implementation is missing the data for the last TR in the data set. Moving all related calculations to a single place is more robust.